lib/pbench/server/api/resources/users_api.py

```diff
@@ -469,20 +470,23 @@ def delete(self, username):
             abort(403, message="Not authorized to delete user")
 
         try:
-            user = User.query(username=username)
-            # Do not delete if the user is admin
-            if not user.is_admin():
-                User.delete(username)
+            User.delete(username)
+            self.logger.info("User entry deleted for user with username {}", username)
+        except NoResultFound:
+            self.logger.info(
+                "Delete operation on non-existent user '{}'", username,
+            )
+            abort(404, message="User not found")
+        except AdminUserDeletion:
+            self.logger.info(
+                "Delete operation on an admin user '{}'", username,
+            )
+            abort(403, message=f"User {username} can not be deletd")
```

Comment:
`s/deletd/deleted`

Reply:
For new code, we should be using the `HTTPStatus.FORBIDDEN` style references.